tests/network/test_services.py

```diff
@@ -55,6 +60,111 @@ class MockBackend(NetworkAPI):
+    def __getattr__(self, rpc_method):
+        return MockRPCMethod(rpc_method, None)
+
+
+class MockRPCMethod(RPCMethod):
+    def __call__(self, *args):
+        if self._rpc_method == "getreceivedbyaddress":
+            assert (
+                args[0] == MAIN_ADDRESS_USED1
+                or args[0] == TEST_ADDRESS_USED2
+                or args[0] == MAIN_ADDRESS_UNUSED
+                or args[0] == TEST_ADDRESS_UNUSED
+            )
+            assert args[1] == 0
+
+            if args[0] in (MAIN_ADDRESS_USED1, TEST_ADDRESS_USED2):
+                return 1.23456789
+            return 0
+
+        if self._rpc_method == "listreceivedbyaddress":
+            assert args[0] == 0
+            assert args[1] is True
+            assert args[2] is True
+            assert (
+                args[3] == MAIN_ADDRESS_USED1
+                or args[3] == TEST_ADDRESS_USED2
+                or args[3] == MAIN_ADDRESS_UNUSED
+                or args[3] == TEST_ADDRESS_UNUSED
+            )
+
+            if args[3] == MAIN_ADDRESS_USED1:
+                return [{
+                    "involvesWatchonly": True,
+                    "address": "1L2JsXHPMYuAa9ugvHGLwkdstCPUDemNCf",
+                    "amount": 1.23456789,
+                    "confirmations": 0,
+                    "label": "",
+                    "txids": [
+                      "381f1605dd927151fbfac2e88608464414fa5b01bd6298cd1e2d9d991907aa9e",
+                      "6e05c708d88cc5bf0f1533938c969de2cc48f438b0ae28ce89fefbaa1938185a"
+                    ]}]
+            elif args[3] == TEST_ADDRESS_USED2:
+                return [{
+                    "involvesWatchonly": True,
+                    "address": "mmvP3mTe53qxHdPqXEvdu8WdC7GfQ2vmx5",
+                    "amount": 1.23456789,
+                    "confirmations": 0,
+                    "label": "",
+                    "txids": [
+                      "ff2b4641481f1ee553ba2c9f02f413a86f70240c35b5aee554f84e3efee93292",
+                      "ef9bd3ac4eacc60a16117eaca0631bbef673eb8a71084de4b9ada3317f7895e9",
+                      "ff2b4641481f1ee553ba2c9f02f413a86f70240c35b5aee554f84e3efee93290"
+                    ]}]
+            return []
+
+        if self._rpc_method == "getrawtransaction":
+            assert args[0] == MAIN_TX_VALID or args[0] == TEST_TX_VALID
+            assert args[1] is False
+            return True
+
+        if self._rpc_method == "listunspent":
+            assert args[0] == 0
+            assert args[1] == 9999999
+            assert (
+                args[2] == [MAIN_ADDRESS_USED1]
+                or args[2] == [TEST_ADDRESS_USED2]
+                or args[2] == [MAIN_ADDRESS_UNUSED]
+                or args[2] == [TEST_ADDRESS_UNUSED]
+            )
+
+            if args[2][0] in (MAIN_ADDRESS_UNUSED, TEST_ADDRESS_UNUSED):
+                return []
+            return [{
+                "txid": "381f1605dd927151fbfac2e88608464414fa5b01bd6298cd1e2d9d991907aa9e",
```

Comment:
These are fine, but like with my earlier comment about constants, I think we have samples with a lot of these.

I guess if they're used only once there's no reason to add them as samples.

Reply:
I agree! However, transaction constants are found inside `test_transaction.py`, not `samples.py`. And I think we should probably generally clean up the sample constants. Over time we may have even introduced redundant constants, etc. Maybe this is something that should be done in a future PR, which cleans up the sample constants?

Reply:
Sure, that's fine.